### ISSUE TITLE
analytics function doesn't send `t` object

### ISSUE THREAD

**Issue by rstacruz** [MoOx/pjax#43]
Using v0.1.4, I'm getting this error:

``` js
                this.options.analytics = this.options.analytics || function(t) {
                    window._gaq && _gaq.push(["_trackPageview"]),
                    window.ga && ga("send", "pageview", {
                        page: t.url, # Uncaught TypeError: Cannot read property 'url' of undefined
                        title: t.title
                    })
                }
```

**Comment by MoOx** [comment]
where do you find this `t` thing ? I am not using this lib anymore and struggle to find something about it, sorry :/

**Comment by rstacruz** [comment]
It's from here: https://github.com/MoOx/pjax/blob/6eafcf8dc61742adfb0bc7a0f401a849366f1422/lib/proto/parse-options.js#L10-L20

**Comment by MoOx** [comment]
From what I can read, indeed it's not send. Not sure why I did that :(

**Comment by MoOx** [comment]
This lib wasn't tested before 0.2, but it seems 0.2 is not working and nobody wants to contribute :/

**Comment by rstacruz** [comment]
any pointers on how we can fix 0.2?

**Comment by MoOx** [comment]
I "just" splited the code into small pieces, in order to facilitate testing. It seems tests are broken and I think I didn't finish a test or something like that. Should not be a big deal, but "we" just need to find what is going wrong :/

**Comment by MoOx** [comment]
Maybe you can try "npm install & npm test" and see what happens? :D

**Comment by MoOx** [comment]
I can add you as a collab if you want

**Comment by rstacruz** [comment]
it breaks because one test has `t.fail` in it:

https://github.com/MoOx/pjax/blob/master/tests/lib/switch-selectors.js#L6

no indication as to why it's made to fail though.

**Comment by darylteo** [comment]
It's made to fail because there is no test written for it yet. 

Switch Selectors also seems to be broken with issue #46 so I will look to fix that one and implement tests for it.

**Comment by MoOx** [comment]
So is this fixed now (sorry I am lazy)?

**Comment by darylteo** [comment]
No I only fixed the failing test.

Not sure what the issue is in this one because the line of code referenced
doesn't seem to exist in the code base anymore.

**Comment by rstacruz** [comment]
@moox, on a side note, may i ask why you're not using this library anymore?

for me: the alternatives are to use jquery.pjax and wiselinks, both of which rely on jQuery, which I'd prefer not to be a dependency.

**Comment by MoOx** [comment]
I now only build webapps with React (for the last two years) :)

**Comment by rstacruz** [comment]
ah, haha, right

**Comment by darylteo** [comment]
What about us poor legacy maintainers ...

*grumbles*

**Comment by evildancer** [comment]
this.options.analytics = this.options.analytics || function(t) {
                    window._gaq && _gaq.push(["_trackPageview"]),
                    window.ga && ga("send", "pageview", {
                        page: t.url, # Uncaught TypeError: Cannot read property 'url' of undefined
                        title: t.title
                    })
                }

**Comment by BehindTheMath** [comment]
This should be fixed by #59.
